File: data_format_transform.cpp
#include<iostream>
#include<math.h>
using namespace std;

int main() {
	int num = 0; //number of q-points
	cin >> num;
	double temp1 = 0, temp2 = 0, temp3 = 0;
	double qx = 0, qy = 0, qz = 0;
	double qx1 = 0, qy1 = 0, qz1 = 0;
	double q[500] = {}; //coordinate of q
	double freq[500][9] = {};  //phonon frequencies
	int i = 0, j = 0;
	//input
	//copy the context of the freq file
	for (i = 0; i < num; i++) { 
		cin >> qx >> qy >> qz;
		for (j = 0; j < 9; j++) {
			cin >> temp1 >> temp2 >> freq[i][j] >> temp3;
		}
		double temp = (qx-qx1)*(qx-qx1)+ (qy - qy1) * (qy - qy1)+ (qz - qz1) * (qz - qz1);
		if (i != 0)
			q[i] = sqrt(temp) + q[i - 1];
		else
			q[i] = sqrt(temp);
		qx1 = qx, qy1 = qy, qz1 = qz;
	}
	//output
	for (i = 0; i < num; i++) {
		cout << q[i];
		for (j = 0; j < 9; j++) {
			cout << " " << freq[i][j];
		}
		cout << endl;
	}
	return 0;
}

File: data_format_transform.py
import math

num = int(input("Enter the number of q-points: "))
temp1, temp2, temp3 = 0, 0, 0
qx, qy, qz = 0, 0, 0
qx1, qy1, qz1 = 0, 0, 0
q = [0.0] * 500  # coordinate of q
freq = [[0.0] * 9 for _ in range(500)]  # phonon frequencies

# Input - Copy the content of the freq file
for i in range(num):
    qx, qy, qz = map(float, input().split())
    for j in range(9):
        temp1, temp2, freq[i][j], temp3 = map(float, input().split())
    
    temp = (qx - qx1) ** 2 + (qy - qy1) ** 2 + (qz - qz1) ** 2
    if i != 0:
        q[i] = math.sqrt(temp) + q[i - 1]
    else:
        q[i] = math.sqrt(temp)
    qx1, qy1, qz1 = qx, qy, qz

# Output
for i in range(num):
    print(q[i], end=' ')
    for j in range(9):
        print(freq[i][j], end=' ')
    print()

